pmd-core/src/main/java/net/sourceforge/pmd/util/log/AntLogHandler.java

```diff
@@ -23,10 +25,52 @@
     private Project project;
 
     private static final Formatter FORMATTER = new PmdLogFormatter();
+    private static final Level[] LOG_LEVELS = new Level[] {
+        Level.SEVERE, Level.WARNING, Level.INFO, Level.CONFIG, Level.FINEST,
+    };
 
     public AntLogHandler(Project project) {
         this.project = project;
     }
+    
+    public Level getAntLogLevel() {
+        Class<?> clazz;
+        int maxLevel = Project.MSG_ERR;
+        for (final BuildListener l : project.getBuildListeners()) {
+            clazz = l.getClass();
+            Field declaredField = null;
+            try {
+                while (BuildListener.class.isAssignableFrom(clazz)) {
+                    try {
+                        declaredField = l.getClass().getDeclaredField("msgOutputLevel");
```

Comment:
This is broken. The `l.getClass()` should be `clazz`.

Btw you may be interested in Apache's `FieldUtils`, eg `getDeclaredField` and `readField` methods, which makes this kind of logic more concise.

Reply:
Thanks. I discarded `FieldUtils` since I need to look for 2 separate fields and wanted to avoid transversing the hierarchy twice to do so. Also, I know I don't need to check interfaces if not found. I know it's probably early optimization, but non the less.

Non the less, I'm most interested in feedback on the hacky approach / alternatives for the time being.